src/target.zig

```diff
@@ -487,6 +487,35 @@ pub fn hasRedZone(target: std.Target) bool {
 pub fn libcFullLinkFlags(target: std.Target) []const []const u8 {
     // The linking order of these is significant and should match the order other
     // c compilers such as gcc or clang use.
+
+    // Starting with glibc 2.28, we need to link "c_nonshared" for "pthread_atfork" function
```

Comment:
Is there any harm to linking this for older glibc versions? how long has this `c_nonshared` library existed?

Reply:
> how long has this c_nonshared library existed?

https://askubuntu.com/questions/497306/cannot-compile-gcc-or-g-libc-nonshared-a-not-recognized 
Jul 13, 2014 

> Would linking this unconditionally break anything for glibc versions?

AFAIK default linker script has been doing this since ancient times.